lib/include/openturns/OTtestcode.hxx

```diff
@@ -379,6 +380,17 @@ inline void assert_almost_equal(const SymmetricMatrix &a, const SymmetricMatrix
   }
 }
 
+inline void assert_almost_equal(const Distribution &a, const Distribution &b, const Scalar rtol = 1.0e-5, const Scalar atol = 1.0e-8, const String errMsg = "")
+{
+  if (a.getImplementation()->getClassName() != b.getImplementation()->getClassName())
+    throw InvalidArgumentException(HERE) << "A and B must be the same distribution. A is a " << a.getImplementation()->getClassName() << " whereas B is a " << b.getImplementation()->getClassName();
+
+  if (a.getParameterDimension() != b.getParameterDimension())
+    throw InvalidArgumentException(HERE) << "A and B must have the same number of parameters. A has " << a.getParameterDimension() << " parameters whereas B has " << b.getParameterDimension() << " parameters.";
+
+  assert_almost_equal(a.getParameter(), b.getParameter(), rtol, atol, errMsg);
+}
```

Comment:
Wrong. The WeibullMax and GeneralizedExtremeValue distributions have the same number of parameters but are different if the parameters are equal

Reply:
The assert would throw in this case because of the class name check above.